spaceship/lib/spaceship/tunes/iap.rb

```diff
@@ -0,0 +1,112 @@
+      #
+      #   }
+      # }
+      # @param reference_name (String): iTC Reference Name
+      # @param product_id (String): A unique ID for your in-app-purchase
+      # @param bundle_id (String): The bundle ID must match the one you used in Xcode. It
+      # @param cleared_for_sale (Boolean): Is this In-App-Purchase Cleared for Sale
+      # @param review_notes (String): Review Notes
+      # @param review_screenshot (String): Path to the screenshot (should be 640x940 PNG)
+      # @param pricing_intervals (Hash): a Hash of the languages
+      # @example:
+      #  [
+      #    {
+      #      country: "WW",
+      #      begin_date: nil,
+      #      end_date: nil,
+      #      tier: 1
+      #    }
+      #  ]
+      # @param family_id (String) Only used on RECURRING purchases, assigns the In-App-Purchase to a specific familie
+      # @param subscription_free_trial (String) Free Trial duration (1w,1m,3m....)
+      # @param subscription_duration (String) 1w,1m.....
+      def create!(type: "consumable",
+                  versions: nil,
+                  reference_name: nil,
+                  product_id: nil,
+                  cleared_for_sale: true,
+                  review_notes: nil,
+                  review_screenshot: nil,
+                  pricing_intervals: nil,
+                  family_id: nil,
+                  subscription_free_trial: nil,
+                  subscription_duration: nil)
+
+        client.create_iap!(app_id: self.application.apple_id,
+                           type: type,
+                           versions: versions,
+                           reference_name: reference_name,
+                           product_id: product_id,
+                           cleared_for_sale: cleared_for_sale,
+                           review_notes: review_notes,
+                           review_screenshot: review_screenshot,
+                           pricing_intervals: pricing_intervals,
+                           family_id: family_id,
+                           subscription_duration: subscription_duration,
+                           subscription_free_trial: subscription_free_trial)
+      end
+
+      # find a specific product
+      # @param product_id (String) Product Id
+      def find(product_id)
+        all.each do |product|
+          if product.product_id == product_id
+            return product
+          end
+        end
+        return nil
+      end
+
+      # return all available In-App-Purchase's of current app
+      # this is not paged inside iTC-Api so if you have alot if IAP's (2k+)
+      # it might take some time to load
+      def all
+        r = client.iaps(app_id: self.application.apple_id)
+        return_iaps = []
+        r.each  do |product|
+          attrs = product
+          attrs[:application] = self.application
+          loaded_iap = Tunes::IAPList.factory(attrs)
+          next if loaded_iap.status == "deleted"
```

Comment:
Is there any case when we might want to receive `deleted`? Just asking... I don't know myself